### DISCUSSION TITLE
Pumping analysis of a district heating network

### DISCUSSION THREAD

**Discussion by GianmarcoPreso** [OpenModelica/OpenModelica#10674] [Q&A]
hello everyone!

i'm trying to model a district heating network, focusing on the fluidynamic problem. How you see from the attached image, my network is modeled in a pretty simple way. We have:
- Static Pipes
- Users: modeled with a constant mass flow rate pump
- Repumping station: modeled with constant dp
- Supply Pumps: constant mass flow rate

![DH](https://github.com/OpenModelica/OpenModelica/assets/125891549/74ea9f21-27bf-45d3-8f47-45b33555e828)

The problem pops up when i add an other supply pump: when the network is supplied just by Pump1 and Pump2, the simulation finish successfully; when i add the Pump3 and Pump4, openmodelica is not able to solve the problem. I'm setting every pump with a mass low rate = TotalMassFlow/4.

The error is always this:

The following assertion has been violated at time 1.000000
noEvent(csvWriPB.u[11] >= 0.0)
Pressure (= -55071.4 Pa) of medium "Buildings.Media.Water" is negative
(Temperature = 293.235 K)
The following assertion has been violated at time 1.000000
csvWriPB.u[11] >= 0.0 and csvWriPB.u[11] <= 100000000.0
Variable violating min/max constraint: 0.0 <= csvWriPB.u[11] <= 100000000.0, has value: -55071.4
No event found, but assert was triggered. Throwing now!

I've tried any modification at the numbers, but i'm not finding the solution.

**Comment by casella** [comment]
@GianmarcoPreso I understand your system boundaries are entirely made of prescribed mass flow rates. This is not going to end well, if there is a slightly mismatch between the total supply and the total consumption, given the very low compressibility of water, the pressure will very quickly go through the roof. Do I miss something?

**Reply by GianmarcoPreso** [reply]
@casella Thanks for the answer. You got the problem, all the system boundaries are prescribed mass flow rate, but i need to work in this way because in a district heating network you have to impose a mass flow rate at the outlet of the pumping systems, and then through the prevalence of the network, the pumps work at a certain pressure. Seen your massive experience with modelica, i would ask you:
- How you can see, i've put BC on the top left of the system. Does the position of the BC influence the convergence of the solution?
- The system work perfectly if i put only the pump1 and pump2. How can it be possible?
- In general, you have any suggestion?

Sorry if i bother you. Thanks in advance.

**Comment by GianmarcoPreso** [comment]
Hi everyone!
I'm going to close this discussion because the problem has been solved. It was just a problem of pressure of the boundary.

Thank you all!

**Comment by casella** [comment]
As a side note, you always need a pressure source in incompressible fluid networks. Eventually, no flow will get into or out of it (or almost none, if the fluid is nearly incompressible), but that is necessary to set the pressure of the whole thing.

This role is usually played by pressurized accumulators, or possibly by connections to a municipal water supply.

**Reply by GianmarcoPreso** [reply]
Thank you Prof. Casella. The problem was exactly that, my pressure source was too low (1 bar), so at a certain point the pressure in the network was negative.
